pkg/block/azure/client_cache.go

```diff
@@ -13,12 +13,14 @@ import (
 )
 
 type ClientContainerCache struct {
+	serviceToClient   *xsync.MapOf[string, *service.Client]
 	containerToClient *xsync.MapOf[string, *container.Client]
 	params            params.Azure
 }
 
 func NewCache(p params.Azure) *ClientContainerCache {
 	return &ClientContainerCache{
+		serviceToClient:   xsync.NewMapOf[*service.Client](),
```

Comment:
If you know - why didn't we use sync.Map?

Reply:
I just followed the convention - maybe @itaiad200 can provide more information

Reply:
Me 🤝 golang generics